src/mpi/coll/bcast/bcast_allcomm_nb.c

```diff
@@ -17,7 +17,10 @@ int MPIR_Bcast_allcomm_nb(void *buffer, MPI_Aint count, MPI_Datatype datatype, i
 
     mpi_errno = MPIC_Wait(req_ptr);
     MPIR_ERR_CHECK(mpi_errno);
+
+    mpi_errno = req_ptr->status.MPI_ERROR;
     MPIR_Request_free(req_ptr);
+    MPIR_ERR_CHECK(mpi_errno);
```

Comment:
Is this necessary? I believe `MPIC_Wait` will return `MPI_ERR_IN_STATUS` if `req_ptr->status.MPI_ERROR` is not 0.

Reply:
I see. Looks like we didn't get your commit 46921feda06ac29db9d2c6ffe71e6646ea9d13ef, where mpi_errno is set from req_ptr->status.

Reply:
removed this commit.